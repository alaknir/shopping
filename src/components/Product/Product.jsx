import React from "react";

import ProductSpeciality from "../ProductSpeciality";

const Product = ({ imgSrc, productName, isSale, isExclusive, price }) => (
  <div className="product">
    <img src={imgSrc} alt={productName} className="product__product-img" />
    <ProductSpeciality isSale={isSale} isExclusive={isExclusive} />
    <div>
      <span className="product__product-name">{productName}</span>
      <span className="product__product-price">{price}</span>
    </div>
  </div>
);

export default Product;
