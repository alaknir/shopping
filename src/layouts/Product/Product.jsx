import React from "react";
import ProductContainer from "../../containers/Product";

const Products = () => <ProductContainer />;

export default Products;
