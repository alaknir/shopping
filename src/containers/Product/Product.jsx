import React from "react";
import { withRouter, Redirect } from "react-router-dom";
import { connect } from "react-redux";

import Product from "../../components/Product";
import ProductSize from "../../components/ProductSize";
import ProductsHeader from "../../components/ProductsHeader";

const ProductContainer = ({ products, match }) => {
  if (!products || !products.length) {
    return <Redirect to="/products" />;
  }

  const product = products.filter(product => match.params.id === product.id);
  const {
    id,
    isExclusive,
    isSale,
    price,
    productImage,
    productName,
    size
  } = product[0];
  return (
    <div className="row col-sm-4 product-container">
      <ProductsHeader heading={productName} />
      <Product
        {...{
          id,
          isExclusive,
          isSale,
          price,
          productName,
          size
        }}
        imgSrc={`../assets/products/${productImage}`}
      />
      <div className="clearfix" />
      <div>
        <ProductSize productId={id} sizeList={size} />
      </div>
    </div>
  );
};

const mapStateToProps = state => ({
  products: state.products.data
});

export default connect(mapStateToProps)(withRouter(ProductContainer));
