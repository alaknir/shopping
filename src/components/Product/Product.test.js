import React from "react";
import { mount } from "enzyme";

import Product from "./Product";

describe("Product container", () => {
  it("Should render properly", () => {
    const component = (
      <Product
        {...{
          id: "12_34",
          imgSrc: "",
          productName: "test1",
          isSale: true,
          isExclusive: true,
          price: "$15.00",
          size: ["XS", "S", "M", "L", "XL"]
        }}
      />
    );
    const wrapper = mount(component);
    expect(wrapper).toMatchSnapshot();
  });
});
