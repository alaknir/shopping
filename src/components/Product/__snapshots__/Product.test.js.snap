// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product container Should render properly 1`] = `
<Product
  id="12_34"
  imgSrc=""
  isExclusive={true}
  isSale={true}
  price="$15.00"
  productName="test1"
  size={
    Array [
      "XS",
      "S",
      "M",
      "L",
      "XL",
    ]
  }
>
  <div
    className="product"
  >
    <img
      alt="test1"
      className="product__product-img"
      src=""
    />
    <ProductSpeciality
      isExclusive={true}
      isSale={true}
    >
      <div
        className="row product-spl"
      >
        <div
          className="col-xs-12 product-spl__badge"
        >
          <span
            className="product-spl__badge-sale"
          >
            Sale
          </span>
          <span
            className="product-spl__badge-exclusive"
          >
            Exclusive
          </span>
        </div>
      </div>
    </ProductSpeciality>
    <div>
      <span
        className="product__product-name"
      >
        test1
      </span>
      <span
        className="product__product-price"
      >
        $15.00
      </span>
    </div>
  </div>
</Product>
`;
